share/completions/zfs.fish

```diff
@@ -0,0 +1,479 @@
+# Fish completions for the OpenZFS zfs command
+# Possible enhancements:
+# - add a test to propose iSCSI and Trusted Extensions completions only when such system is present;
+# - Illumos man pages suggests that it does not support nbmand nor atime mount option, so these properties should be proposed only when available
+# - generally, propose properties only when the current OS and ZFS versions support them;
+# - for the promote command, propose only eligible filesystems;
+# - for the rollback command, propose only the most recent snapshot for each dataset, as it will not accept an intermediary snapshot;
+# - for the release command, complete with existing tags;
+# - for the diff command, complete the destination dataset of the diff;
+# - for the program command, complete the script to be executed
+# - for commands accepting several arguments of different types, propose arguments in the right order: for get, once the ZFS parameters have been given, only propose datasets
+
+set OS ""
+if passwd --help >/dev/null ^&1
+    set OS "Linux"
+else # Not Linux, so use the ugly uname
+    set -l os_type (uname)
+    switch $os_type
+        case Darwin
+            set OS "macOS"
+        case FreeBSD
+            set OS "FreeBSD"
+        case SunOS
+            set OS "SunOS"
+        # Others?
+        case "*"
+            set OS "unknown"
+    end
+end
+
+function __fish_zfs_append -d "Internal completion function for appending string to the ZFS commandline"
+    set str (commandline -tc| sed -ne "s/\(.*,\)[^,]*/\1/p"|sed -e "s/--.*=//")
```

Comment:
I'll try that. Note that [`__fish_append`](https://github.com/fish-shell/fish-shell/blob/master/share/functions/__fish_append.fish) could be corrected as well, as I essentially borrowed its code and stripped off what was useless to me.

Reply:
In fact, it would be a bit difficult to test that, as my current config would not allow me to easily test that; as you know fish way better than me, I assume you are right and blindly replaced my code with yours.